alf/environments/igibson/suite_igibson.py

```diff
@@ -0,0 +1,384 @@
+        observation.pop('rgb')
+        observation.pop('depth')
+        observation['rgbd'] = rgbd_obs
+        return observation
+
+
+@alf.configurable()
+class OneHotIntWrapper(BaseObservationWrapper):
+    """Transforms the data type of one-hot input from float to int"""
+
+    def __init__(self, env, fields=None):
+        super().__init__(env, fields=fields)
+
+    def transform_space(self, observation_space):
+        obs_low = np.full_like(
+            observation_space.low, fill_value=0, dtype=np.uint8)
+        obs_high = np.full_like(
+            observation_space.high, fill_value=255, dtype=np.uint8)
+        assert obs_low.shape == obs_high.shape
+        observation_space = gym.spaces.Box(
+            low=obs_low, high=obs_high, dtype=np.uint8)
+        return observation_space
+
+    def transform_observation(self, observation):
+        return (observation * 255).astype(np.uint8)
+
+
+@alf.configurable()
+class ImageIntWrapper(BaseObservationWrapper):
+    """Transforms the data type of image input from float to int"""
+
+    def __init__(self, env, fields=None):
+        super().__init__(env, fields=fields)
+
+    def transform_space(self, observation_space):
+        obs_low = np.full_like(
+            observation_space.low, fill_value=0, dtype=np.uint8)
+        obs_high = np.full_like(
+            observation_space.high, fill_value=255, dtype=np.uint8)
+        assert obs_low.shape == obs_high.shape
+        observation_space = gym.spaces.Box(
+            low=obs_low, high=obs_high, dtype=np.uint8)
+        return observation_space
+
+    def transform_observation(self, observation):
+        return (observation * 255).astype(np.uint8)
+
+
+@alf.configurable()
+class RenderWrapper(gym.Wrapper):
+    """Implements render() for iGibson env for visualization purposes"""
+
+    def __init__(self, env):
+        super().__init__(env)
+        self.env = env
+        self.metadata['render.modes'] = ['rgb_array', 'human']
+
+    def render(self, mode='human', **kwargs):
+        if 'vision' in self.env.sensors:
+            vision_obs = self.sensors['vision'].get_obs(self)
+            if 'rgb' in vision_obs:
+                rgb_obs = vision_obs['rgb']
+                rgb_obs = (rgb_obs * 255).astype(np.uint8)
+                if 'depth' not in vision_obs:
+                    return rgb_obs
+                else:
+                    depth_obs = vision_obs['depth']
+                    depth_obs = (depth_obs * 255).astype(np.uint8)
+                    depth_obs_rescaled = np.tile(depth_obs, (1, 1, 3))
+                    rgbd_obs = np.concatenate((rgb_obs, depth_obs_rescaled),
+                                              axis=1)
+                    return rgbd_obs
+            else:
+                raise AttributeError("robot doesn't have modality: rgb")
+        else:
+            raise AttributeError("robot doesn't have sensor: vision")
+
+
+@alf.configurable()
+class GFTTupleInputWrapper(gym.ObservationWrapper):
```

Comment:
Seems not quite necessary? Before calling GFT, if the `time_step.observation` is a nest (any structure including dict), you can first do `alf.nest.flatten(time_step.observation)` to convert it to a list/tuple.